### DISCUSSION TITLE
Is navigation-tree still available?

### DISCUSSION THREAD

**Discussion by msharp1** [datenstrom/community#507] [Ask a question] [answered]
hi @markseu  I've been trying Yellow five years ago and reported datenstrom/community#123 at that time.
I want to try again building a website on top of yellow. I wanted to know if the navigation-tree was still available in yellow ? I'm unable to find any reference about it in documentation nor the source files in the repo. Has it been removed on purpose ?
Cheers

**Comment by BsNoSi** [comment]
In the meantime Yellow has developed considerably further. One general change took place in theme handling. Download current yellow and give it a try.

What are you looking for concretly? There are several different navigation solutions available. As you noticed correctly in datenstrom/community#123 a droping menu is not the best idea for mobile or tablet.

You may have a look on my home page ( https://nosi.de) which operates without java script only with a very small css file to deal this. If there is enough space you have e head navigation with subentries on the right hand, small screens have a drop down menu (icon on the left side).

I know that it *could be better* but this works fine for me since a few years.

**Comment by msharp1** [comment]
Hi @BsNoSi ! thank you for the answer. The navigation you have on nosi.de would be enough for me. Is it a yellow extension ?

**Comment by BsNoSi** [comment]
No extension. It is just a layout. Please give me some days to extract it in a sample for reuse. I will post a note here.

**Comment by BsNoSi** [comment, marked as answer]
@msharp1:  This is the `navigation.html` of [NoSi.de](https://nosi.de) doing the whole job (plus some css of your personal taste):
```
<?php $pages = $this->yellow->content->top() ?>
<?php $this->yellow->page->setLastModified($pages->getModified()) ?>
 
 <nav id="navi1" class="navi">
<div class="navigation" role="navigation">
<?php foreach ($pages as $page): ?>
<div><a<?php echo $page->isActive() ? " class=\"active\" aria-current=\"page\"" : "" ?> href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("titleNavigation") ?></a></div>
<?php endforeach ?>
</div>
<div class="sub">
<?php
$pages = $this->yellow->page->getChildren($showInvisible = false);
  if( ($this->yellow->page->getParent()) AND (count($pages)==0))  {  $pages = $this->yellow->page->getSiblings($showInvisible = false);   }
   if(count($pages))  {
	foreach($pages as $page) {				  
        $title = $page->get("titleNavigation");
        $curr = $page -> isActive() ? ' class="active" aria-current="page"' : '';
		echo '<div><a ' . $curr . 'href="' . $page->getLocation(true) . '">' . htmlspecialchars($title) . '</a></div>';
	 }
  } 
?>
</div>
</nav>
<div class="navigation-banner"></div>
```

Be aware that all pages with this navigation generate a tree of all files in the (sub)folder.

This code may be improved with  `Status` check to aknowledge `public`, `unlisted`, `draft` (or whatever) to avoid showing files you don't want to show etc. 

> Please, be so kind and share any improvements with the community.

**Comment by msharp1** [comment]
Thank you very much @BsNoSi ! It's very kind of you. Let's close the issue.
